imports/client/components/PuzzlePage.tsx

```diff
@@ -1279,11 +1279,15 @@ const PuzzlePage = React.memo(() => {
       Documents.findOne({ puzzle: puzzleId }, { sort: { createdAt: 1 } })
   ), [puzzleDataLoading, puzzleId]);
 
-  const activePuzzle = useTracker(() => Puzzles.findOneAllowingDeleted(puzzleId), [puzzleId]);
+  const activePuzzle = useTracker(() => (
+    puzzleDataLoading ?
+      undefined :
+      Puzzles.findOneAllowingDeleted(puzzleId)
+  ), [puzzleDataLoading, puzzleId]);
   const breadcrumbTitle =
     puzzleDataLoading ?
       'loading...' :
-      `${activePuzzle!.title}${activePuzzle!.deleted ? ' (deleted)' : ''}`;
+      `${activePuzzle?.title || '(no title)'}${activePuzzle?.deleted ? ' (deleted)' : ''}`;
```

Comment:
OK, now that I understand what's going on, I think that I would be slightly happier if this were instead

```
  const breadcrumbTitle =
    activePuzzle ?
      'loading...' :
      `${activePuzzle.title}${activePuzzle.deleted ? ' (deleted)' : ''}`;
```

and if additionally we got rid of the `!` operator on line 1344 or so (passed to `PuzzlePageMetadata`), probably by making the conditional before it into `puzzleDataLoading || !activePuzzle`.

We should ultimately have better handling for 404-like errors than we do now, but we don't need to shave that yak here.

Reply:
I decided to do at least a modicum of shaving that yak since I was unhappy with having an incorrect loading error, and I felt it would obscure the actual logic/constraints.